warp/yul/ToCairoVisitor.py

```diff
@@ -38,24 +45,38 @@
     "range_check_ptr": None,
     "storage_ptr": "Storage*",
     "syscall_ptr": "felt*",
+    "exec_env": "ExecutionEnvironment",
 }
 
 IMPLICITS_SET = set(IMPLICITS.keys())
 
 
 class ToCairoVisitor(AstVisitor):
-    def __init__(self, sol_source: str, public_functions: list[str]):
+    def __init__(self, sol_source: str, sol_src_path: str, main_contract: str):
```

Comment:
This change seems to be unnecessary?.. We don't use `sol_src_path` and `main_contract` for anything other than computing `public_functions`

Reply:
we write the name of main_contract to a file so we can use it later for cli actions like invoke.